joomla/joomla-cms
# Add typehinting to Actionlog

### Summary of Changes
Since we are supporting PHP 7.2+ in Joomla 4, we should add typehinting for more robust code. This PR adds typehinting and cleans up some code for the com_actionlog component.

There are no new features, only code cleanup.

### Testing Instructions
1. Apply this PR
2. Check in the plugins if the actionlog plugins are enabled
3. Check the following actions work without error and are recorded in the actionlogs:
- [ ] Saving an article
- [ ] Deleting an article
- [ ] Change state of an article
- [ ] Saving the global configuration
- [ ] Install an extension
- [ ] Uninstall an extension
- [ ] Update an extension
- [ ] Save plugin settings
- [ ] Add a module
- [ ] Delete a module
- [ ] Create a user
- [ ] Delete a user
- [ ] Create a usergroup
- [ ] Delete a usergroup
- [ ] User normal login
- [ ] User failed login
- [ ] User logout
- [ ] Send password reminder
- [ ] Send username reminder
- [ ] Check-in an article
- [ ] Export action logs
- [ ] Purge the action logs

### Expected result
All actions are successful and recorded in the action logs

### Documentation Changes Required
None as far as I am aware.
